app/screens/content-view-screen/content-view-screen.tsx

```diff
@@ -55,23 +56,64 @@ export interface ContentViewNavigationStateParams {
 }
 export interface ContentViewScreenProps extends NavigationScreenProps<ContentViewNavigationStateParams> {}
 
+function delayed(callback: () => void, delay: number) {
+  let nextTime = 0
+  return () => {
+    const now = Date.now()
+    if (!nextTime) nextTime = now;
+    const diff = Math.min(delay, Math.max(now - nextTime, 0))
+    setTimeout(callback, diff)
+    nextTime += delay - diff
+  }
+}
+
+const getMattersSetLikeCoinButtonEnableScript = (isEnabled = true) => {
+  return `
+    if (document.querySelector('.appreciate-button')) {
+      document.querySelector('.appreciate-button').style.pointerEvents = '${isEnabled ? "all" : "none"}';
+    }
+  `;
+}
+
 const injectedJavaScript = `${
   /* Forward any post messages to WebView */ ""
   }window.addEventListener('message', function(event) {
     window.ReactNativeWebView.postMessage(JSON.stringify(event.data));
   });
-  ${/* Hide LikeCoin button on custom site */ ""
+  ${/* Grab meta */ ""
   }window.onload = function () {
-    if (window.location.host === "matters.news") {
-      document.querySelector('.appreciate-button').hidden = true;
+    let meta = {};
+    let host = window.location.host;
+    switch (host) {
+      case 'matters.news':
+        meta.site = 'matters';
+        meta.isLoggedIn = !!document.querySelector('.me-avatar');
```

Comment:
What hapoens if matters changes its ui, what would be broken?

Reply:
No effect on clicking in-app like button